awx/api/views/__init__.py

```diff
@@ -412,7 +412,10 @@ def post(self, request, *args, **kwargs):
 
             execution_node_health_check.apply_async([obj.hostname])
         else:
-            from awx.main.tasks.system import cluster_node_health_check
+            return Response(
+                {"error": f"Cannot run a health check on instances of type {obj.node_type}.  Health checks can only be run on execution nodes."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
 
             cluster_node_health_check.apply_async([obj.hostname], queue=obj.hostname)
```

Comment:
This line (and maybe the method itself) now becomes unreachable and you should be free to remove it.